File: src/data/csv_row.py
class CSVRow:
    def __init__(self, **kwargs):
        self.Country_Name = kwargs.get('Country Name')
        self.Country_Code = kwargs.get('Country Code')
        self.Indicator_Name = kwargs.get('Indicator Name')
        self.Indicator_Code = kwargs.get('Indicator Code')
        self.Year_1960 = kwargs.get('1960')
        self.Year_1961 = kwargs.get('1961')
        self.Year_1962 = kwargs.get('1962')
        self.Year_1963 = kwargs.get('1963')
        self.Year_1964 = kwargs.get('1964')
        self.Year_1965 = kwargs.get('1965')
        self.Year_1966 = kwargs.get('1966')
        self.Year_1967 = kwargs.get('1967')
        self.Year_1968 = kwargs.get('1968')
        self.Year_1969 = kwargs.get('1969')
        self.Year_1970 = kwargs.get('1970')
        self.Year_1971 = kwargs.get('1971')
        self.Year_1972 = kwargs.get('1972')
        self.Year_1973 = kwargs.get('1973')
        self.Year_1974 = kwargs.get('1974')
        self.Year_1975 = kwargs.get('1975')
        self.Year_1976 = kwargs.get('1976')
        self.Year_1977 = kwargs.get('1977')
        self.Year_1978 = kwargs.get('1978')
        self.Year_1979 = kwargs.get('1979')
        self.Year_1980 = kwargs.get('1980')
        self.Year_1981 = kwargs.get('1981')
        self.Year_1982 = kwargs.get('1982')
        self.Year_1983 = kwargs.get('1983')
        self.Year_1984 = kwargs.get('1984')
        self.Year_1985 = kwargs.get('1985')
        self.Year_1986 = kwargs.get('1986')
        self.Year_1987 = kwargs.get('1987')
        self.Year_1988 = kwargs.get('1988')
        self.Year_1989 = kwargs.get('1989')
        self.Year_1990 = kwargs.get('1990')
        self.Year_1991 = kwargs.get('1991')
        self.Year_1992 = kwargs.get('1992')
        self.Year_1993 = kwargs.get('1993')
        self.Year_1994 = kwargs.get('1994')
        self.Year_1995 = kwargs.get('1995')
        self.Year_1996 = kwargs.get('1996')
        self.Year_1997 = kwargs.get('1997')
        self.Year_1998 = kwargs.get('1998')
        self.Year_1999 = kwargs.get('1999')
        self.Year_2000 = kwargs.get('2000')
        self.Year_2001 = kwargs.get('2001')
        self.Year_2002 = kwargs.get('2002')
        self.Year_2003 = kwargs.get('2003')
        self.Year_2004 = kwargs.get('2004')
        self.Year_2005 = kwargs.get('2005')
        self.Year_2006 = kwargs.get('2006')
        self.Year_2007 = kwargs.get('2007')
        self.Year_2008 = kwargs.get('2008')
        self.Year_2009 = kwargs.get('2009')
        self.Year_2010 = kwargs.get('2010')
        self.Year_2011 = kwargs.get('2011')
        self.Year_2012 = kwargs.get('2012')
        self.Year_2013 = kwargs.get('2013')
        self.Year_2014 = kwargs.get('2014')
        self.Year_2015 = kwargs.get('2015')
        self.Year_2016 = kwargs.get('2016')
        self.Year_2017 = kwargs.get('2017')
        self.Year_2018 = kwargs.get('2018')
        self.Year_2019 = kwargs.get('2019')
        self.Year_2020 = kwargs.get('2020')
        self.Year_2021 = kwargs.get('2021')
        self.Year_2022 = kwargs.get('2022')
        self.Year_2023 = kwargs.get('2023')

    def __repr__(self):
        return (f"CSVRow(Country_Name={self.Country_Name}, Country_Code={self.Country_Code}, "
                f"Indicator_Name={self.Indicator_Name}, Indicator_Code={self.Indicator_Code}, "
                f"Year_1960={self.Year_1960}, Year_1961={self.Year_1961}, Year_1962={self.Year_1962}, "
                f"Year_1963={self.Year_1963}, Year_1964={self.Year_1964}, Year_1965={self.Year_1965}, "
                f"Year_1966={self.Year_1966}, Year_1967={self.Year_1967}, Year_1968={self.Year_1968}, "
                f"Year_1969={self.Year_1969}, Year_1970={self.Year_1970}, Year_1971={self.Year_1971}, "
                f"Year_1972={self.Year_1972}, Year_1973={self.Year_1973}, Year_1974={self.Year_1974}, "
                f"Year_1975={self.Year_1975}, Year_1976={self.Year_1976}, Year_1977={self.Year_1977}, "
                f"Year_1978={self.Year_1978}, Year_1979={self.Year_1979}, Year_1980={self.Year_1980}, "
                f"Year_1981={self.Year_1981}, Year_1982={self.Year_1982}, Year_1983={self.Year_1983}, "
                f"Year_1984={self.Year_1984}, Year_1985={self.Year_1985}, Year_1986={self.Year_1986}, "
                f"Year_1987={self.Year_1987}, Year_1988={self.Year_1988}, Year_1989={self.Year_1989}, "
                f"Year_1990={self.Year_1990}, Year_1991={self.Year_1991}, Year_1992={self.Year_1992}, "
                f"Year_1993={self.Year_1993}, Year_1994={self.Year_1994}, Year_1995={self.Year_1995}, "
                f"Year_1996={self.Year_1996}, Year_1997={self.Year_1997}, Year_1998={self.Year_1998}, "
                f"Year_1999={self.Year_1999}, Year_2000={self.Year_2000}, Year_2001={self.Year_2001}, "
                f"Year_2002={self.Year_2002}, Year_2003={self.Year_2003}, Year_2004={self.Year_2004}, "
                f"Year_2005={self.Year_2005}, Year_2006={self.Year_2006}, Year_2007={self.Year_2007}, "
                f"Year_2008={self.Year_2008}, Year_2009={self.Year_2009}, Year_2010={self.Year_2010}, "
                f"Year_2011={self.Year_2011}, Year_2012={self.Year_2012}, Year_2013={self.Year_2013}, "
                f"Year_2014={self.Year_2014}, Year_2015={self.Year_2015}, Year_2016={self.Year_2016}, "
                f"Year_2017={self.Year_2017}, Year_2018={self.Year_2018}, Year_2019={self.Year_2019}, "
                f"Year_2020={self.Year_2020}, Year_2021={self.Year_2021}, Year_2022={self.Year_2022}, "
                f"Year_2023={self.Year_2023})")

File: src/CPlus/data/CSVRow.cpp
#pragma once
#include <string>
#include <iostream>

class CSVRow {
public:
    std::string Country_Name;
    std::string Country_Code;
    std::string Indicator_Name;
    std::string Indicator_Code;
    std::string Year_1960;
    std::string Year_1961;
    std::string Year_1962;
    std::string Year_1963;
    std::string Year_1964;
    std::string Year_1965;
    std::string Year_1966;
    std::string Year_1967;
    std::string Year_1968;
    std::string Year_1969;
    std::string Year_1970;
    std::string Year_1971;
    std::string Year_1972;
    std::string Year_1973;
    std::string Year_1974;
    std::string Year_1975;
    std::string Year_1976;
    std::string Year_1977;
    std::string Year_1978;
    std::string Year_1979;
    std::string Year_1980;
    std::string Year_1981;
    std::string Year_1982;
    std::string Year_1983;
    std::string Year_1984;
    std::string Year_1985;
    std::string Year_1986;
    std::string Year_1987;
    std::string Year_1988;
    std::string Year_1989;
    std::string Year_1990;
    std::string Year_1991;
    std::string Year_1992;
    std::string Year_1993;
    std::string Year_1994;
    std::string Year_1995;
    std::string Year_1996;
    std::string Year_1997;
    std::string Year_1998;
    std::string Year_1999;
    std::string Year_2000;
    std::string Year_2001;
    std::string Year_2002;
    std::string Year_2003;
    std::string Year_2004;
    std::string Year_2005;
    std::string Year_2006;
    std::string Year_2007;
    std::string Year_2008;
    std::string Year_2009;
    std::string Year_2010;
    std::string Year_2011;
    std::string Year_2012;
    std::string Year_2013;
    std::string Year_2014;
    std::string Year_2015;
    std::string Year_2016;
    std::string Year_2017;
    std::string Year_2018;
    std::string Year_2019;
    std::string Year_2020;
    std::string Year_2021;
    std::string Year_2022;
    std::string Year_2023;

    CSVRow() = default;

    // Constructor to initialize from CSV data
    CSVRow(const std::string& countryName, const std::string& countryCode,
           const std::string& indicatorName, const std::string& indicatorCode,
           const std::string& year1960, const std::string& year1961, const std::string& year1962,
           const std::string& year1963, const std::string& year1964, const std::string& year1965,
           const std::string& year1966, const std::string& year1967, const std::string& year1968,
           const std::string& year1969, const std::string& year1970, const std::string& year1971,
           const std::string& year1972, const std::string& year1973, const std::string& year1974,
           const std::string& year1975, const std::string& year1976, const std::string& year1977,
           const std::string& year1978, const std::string& year1979, const std::string& year1980,
           const std::string& year1981, const std::string& year1982, const std::string& year1983,
           const std::string& year1984, const std::string& year1985, const std::string& year1986,
           const std::string& year1987, const std::string& year1988, const std::string& year1989,
           const std::string& year1990, const std::string& year1991, const std::string& year1992,
           const std::string& year1993, const std::string& year1994, const std::string& year1995,
           const std::string& year1996, const std::string& year1997, const std::string& year1998,
           const std::string& year1999, const std::string& year2000, const std::string& year2001,
           const std::string& year2002, const std::string& year2003, const std::string& year2004,
           const std::string& year2005, const std::string& year2006, const std::string& year2007,
           const std::string& year2008, const std::string& year2009, const std::string& year2010,
           const std::string& year2011, const std::string& year2012, const std::string& year2013,
           const std::string& year2014, const std::string& year2015, const std::string& year2016,
           const std::string& year2017, const std::string& year2018, const std::string& year2019,
           const std::string& year2020, const std::string& year2021, const std::string& year2022,
           const std::string& year2023)
        : Country_Name(countryName), Country_Code(countryCode), Indicator_Name(indicatorName),
          Indicator_Code(indicatorCode), Year_1960(year1960), Year_1961(year1961), Year_1962(year1962),
          Year_1963(year1963), Year_1964(year1964), Year_1965(year1965), Year_1966(year1966),
          Year_1967(year1967), Year_1968(year1968), Year_1969(year1969), Year_1970(year1970),
          Year_1971(year1971), Year_1972(year1972), Year_1973(year1973), Year_1974(year1974),
          Year_1975(year1975), Year_1976(year1976), Year_1977(year1977), Year_1978(year1978),
          Year_1979(year1979), Year_1980(year1980), Year_1981(year1981), Year_1982(year1982),
          Year_1983(year1983), Year_1984(year1984), Year_1985(year1985), Year_1986(year1986),
          Year_1987(year1987), Year_1988(year1988), Year_1989(year1989), Year_1990(year1990),
          Year_1991(year1991), Year_1992(year1992), Year_1993(year1993), Year_1994(year1994),
          Year_1995(year1995), Year_1996(year1996), Year_1997(year1997), Year_1998(year1998),
          Year_1999(year1999), Year_2000(year2000), Year_2001(year2001), Year_2002(year2002),
          Year_2003(year2003), Year_2004(year2004), Year_2005(year2005), Year_2006(year2006),
          Year_2007(year2007), Year_2008(year2008), Year_2009(year2009), Year_2010(year2010),
          Year_2011(year2011), Year_2012(year2012), Year_2013(year2013), Year_2014(year2014),
          Year_2015(year2015), Year_2016(year2016), Year_2017(year2017), Year_2018(year2018),
          Year_2019(year2019), Year_2020(year2020), Year_2021(year2021), Year_2022(year2022),
          Year_2023(year2023) {}

    // test printing
    void print() const {
        std::cout << "Country: " << Country_Name << ", Code: " << Country_Code
                  << ", Indicator: " << Indicator_Name << ", Year 1960: " << Year_1960 << std::endl;
    }
};
